ddtrace/contrib/aws_lambda/patch.py

```diff
@@ -32,21 +32,25 @@ def _crash_flush(_, __):
 
 def _handle_signal(sig, f):
     """
-    Wraps the given signal with a previously defined, if exists.
+    Wraps the given signal with a previously defined signal, if
+    it's callable and is different than the function `f` to handle.
+    Else, it doesn't wrap and just sets the function to handle as `f`.
     This to avoid our signals overriding existing ones.
 
-    Returns the handler of the wrapped signal.
+    Returns a signal of type `sig` with function `_f`
     """
-    old_signal = None
-    if callable(signal.getsignal(sig)):
-        old_signal = signal.getsignal(sig)
+    old_signal = signal.getsignal(sig)
+    if not callable(old_signal) or old_signal == f:
+        old_signal = None
```

Comment:
Is this a bug fix? If so we should call it out in a release note.

Reply:
This would only happen in tests, it was strictly modified to make sure everything works when running on a testing environment mimicking an AWS Lambda. 
So, this could be a bug fix which I didn't catch when testing back then. I'll add it in the release note too, thanks!

Reply:
https://github.com/DataDog/dd-trace-py/pull/5190